content/learn/book/storing-data/local-system-param.md

```diff
@@ -1,15 +1,51 @@
 +++
-title = "Local System Parameters"
+title = "System-Local Data Storage"
 insert_anchor_links = "right"
 [extra]
 weight = 5
 status = 'hidden'
 +++
 
-Sometimes you may need to preserve some state between executions of a [system].
-For this, we have the [`Local<T>`] system parameter.
+There's one more place where the ECS can store data: on the systems themselves.
+With the help of the [`Local<T>`] system parameter, you can cache private data from one run of a system to the next.
 
-Any type that implements `Default` can be stored in a local. `Default` is required so that the value can be initialized before the system runs. If your type does not implement `Default`, you can use `Option<T>` instead.
+In most cases, a [resource] is going to be more flexible, as you can read and mutate it from other systems.
+However, from time-to-time, you might want to:
+
+- store some system-specific state about the last value
+- maintain a cache of intermediate results to improve efficiency of computation
+- avoid allocating a new large data structure each frame
+
+Bevy itself uses `Local` system params in two prominent places:
+
+1. As part of the [`EventReader`] abstraction, keeping track of which events each system has read.
+2. In run conditions like [`on_timer`], to track how much time has run.
+
+## Starting values
+
+`Local` system params are always initialized with a default value.
+This value is set by either the [`FromWorld`] trait or the [`Default`] trait.
+The [`FromWorld`] trait allows you to access arbitrary data from the world,
+allowing more complex initialization.
+
+If you can't figure out what a good default value might be, remember that [`Option<T>`]
+implements `Default`, even if `T` does not!
+
+```rust,hide_lines=1
+# use bevy::prelude::*;
+
+struct NoGoodDefaultValue(u8);
+
+fn increment_local_system_data(mut local: Local<Option<NoGoodDefaultValue>>){
+    if local.is_none() {
+        *local = Some(NoGoodDefaultValue(0));
+    }
+    
+    local.unwrap().0 += 1;
```

Comment:
```suggestion
fn increment_local_system_data(mut local: Local<Option<NoGoodDefaultValue>>){
    let local = local.get_or_insert_with(|| {
        NoGoodDefaultValue(0)
    });
    
    local.0 += 1;
```
Death to `unwrap`s!

Reply:
This is better code, but less clear for learners :) The simple if statement does a really good job explaining the pattern, especially to Rust novices.

Reply:
I don't know, I'd say the suggested `get_or_insert_with` is clearer - it's right there in the name :thinking: